src/bin/zmtrainsa

```diff
@@ -117,6 +117,13 @@ fi
 source `dirname $0`/zmshutil || exit 1
 zmsetvars
 
+export TMPDIR=${zmtrainsa_tmp_directory:-${zimbra_tmp_directory}/zmtrainsa}
```

Comment:
I think prefixing with zmtrainsa is fine, perhaps even good :-).

I also agree that the use of 1777 here is potentially a design defect, although there's arguments a few different ways on this (one can certainly argue that our overuse/abuse of the single zimbra account is a pretty bad flaw for example).  I'm not sure we want to get into that here/now though.